### PR TITLE
Additional standardization for users subdirectory

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description

Further discussion with @jsf9k led to agreement that there should be a standard `ProvisionAccount` role for every account.  This PR handles the changes to achieve that.

## 💭 Motivation and Context

We want to make our various accounts as uniform as possible.

## 🧪 Testing

Testing was done by executing a `terraform plan` and verifying that no syntax errors were reported.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: users/README.md [hunk 1]
```diff
@@ -76,25 +76,25 @@ future changes by simply running `terraform apply
...
 | assume_access_terraform_backend_policy_description | The description to associate with the IAM policy that allows assumption of the role with access to the Terraform backend | string | `Allow assumption of the AccessTerraformBackend role in the Terraform account.` | no |
 | assume_access_terraform_backend_policy_name | The name to assign the IAM policy that allows assumption of the role with access to the Terraform backend | string | `Terraform-AssumeAccessTerraformBackend` | no |
-| assume_iam_admin_policy_description | The description to associate with the IAM policy that allows assumption of the role to become an IAM admininistrator | string | `Allow assumption of the IamAdministrator role.` | no |
-| assume_iam_admin_policy_name | The name to assign the IAM policy that allows assumption of the role to become an IAM admininistrator | string | `AssumeIamAdministrator` | no |
+| assume_provision_account_policy_description | The description to associate with the IAM policy that allows assumption of the role to provision all AWS resources in this account | string | `Allow assumption of the ProvisionAccount role.` | no |
```
Comment (jsf9k) [inline users/README.md]: For consistency, this should be `assume_provisionaccount_policy_description`.  See also lines 80, 84, 85 and 97.

File: users/assume_provision_policy.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# IAM assume role policy document that allows assumption of the IAM role that
+# allows full IAM administrator permissions
+data "aws_iam_policy_document" "assume_provision_account_role_doc" {
```
Comment (jsf9k) [inline users/assume_provision_policy.tf]: For consistency, this should be `assume_provisionaccount_role_doc`.  See also line 15.

File: users/provision_role.tf [hunk 3]
```diff
@@ -3,7 +3,7 @@
 # AWS resources in the users account.
 # ------------------------------------------------------------------------------
 
-data "aws_iam_policy_document" "iam_admin_role_assume_role_doc" {
+data "aws_iam_policy_document" "provision_account_role_assume_role_doc" {
```
Comment (jsf9k) [inline users/provision_role.tf]: `provisionaccount_role_assume_role_doc`.  See also line 21.

File: users/variables.tf [hunk 4]
```diff
@@ -33,14 +33,14 @@ variable "assume_access_terraform_backend_policy_name" {
   default     = "Terraform-AssumeAccessTerraformBackend"
 }
 
-variable "assume_iam_admin_policy_description" {
-  description = "The description to associate with the IAM policy that allows assumption of the role to become an IAM admininistrator."
-  default     = "Allow assumption of the IamAdministrator role."
+variable "assume_provision_account_policy_description" {
```
Comment (jsf9k) [inline users/variables.tf]: For consistency, this should be `assume_provisionaccount_policy_description`.  See also lines 41, 61, and 66.

File: users/README.md (hunk 1 above)
Comment (jsf9k) [reply users/README.md]: I did the same thing in the "shared-services" directory in #3!  I must remember incorrectly what we agreed to.

File: users/provision_role.tf (hunk 3 above)
Comment (jsf9k) [reply users/provision_role.tf]: I did the same thing in the "shared-services" directory in #3!  I must remember incorrectly what we agreed to.

File: users/assume_provision_policy.tf (hunk 2 above)
Comment (jsf9k) [reply users/assume_provision_policy.tf]: I did the same thing in the "shared-services" directory in #3!  I must remember incorrectly what we agreed to.

File: users/variables.tf (hunk 4 above)
Comment (jsf9k) [reply users/variables.tf]: I did the same thing in the "shared-services" directory in #3!  I must remember incorrectly what we agreed to.

**Comment by dav3r** [comment]
> We should resolve whether we are going to use `provisionaccount` or `provision_account`. I thought we talked about using the former. One of us will have to change.

Per our discussion @jsf9k, I made the changes to standardize on `provisionaccount` in https://github.com/cisagov/cool-accounts/pull/5/commits/4da6632df38a1c34af26aaa05322c4b228f092da.

**Comment by jsf9k** [comment]
> Per our discussion @jsf9k, I made the changes to standardize on `provisionaccount` in [4da6632](https://github.com/cisagov/cool-accounts/commit/4da6632df38a1c34af26aaa05322c4b228f092da).

Excellent.  I like this better.  See also #7, where I made the analogous changes to the "terraform" and "shared-services" directories.

**Pull request merged**